sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Processor/EventProcessorClientLiveTests.cs

```diff
@@ -139,241 +132,144 @@ await using (var consumer = new EventHubConsumerClient(EventHubConsumerClient.De
-                                            return list;
-                                        }
-                                    )
-                        );
-
-                    eventProcessorManager.AddEventProcessors(1);
-
-                    // Start the event processors.
+                await Task.Delay(500);
+            }
 
-                    await eventProcessorManager.StartAllAsync();
+            // Stop the event processors.
 
-                    // Make sure the event processors have enough time to stabilize.
+            await eventProcessorManager.StopAllAsync();
 
-                    await eventProcessorManager.WaitStabilization();
+            // Validate that we set at least one enqueuedTime
 
-                    // Stop the event processors.
+            Assert.That(enqueuedTime, Is.GreaterThan(DateTimeOffset.MinValue));
 
-                    await eventProcessorManager.StopAllAsync();
+            // Create the event processor manager to manage the event processor which will receive all events FromEnqueuedTime of enqueuedTime.
 
-                    // Validate results.
+            using var secondBatchCancellationSource = new CancellationTokenSource();
+            secondBatchCancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
 
-                    foreach (KeyValuePair<string, List<DateTimeOffset>> kvp in timestamps)
+            var eventProcessorManager2 = new EventProcessorManager
+                (
+                    EventHubConsumerClient.DefaultConsumerGroupName,
+                    connectionString,
+                    onInitialize: eventArgs =>
+                        eventArgs.DefaultStartingPosition = EventPosition.FromEnqueuedTime(enqueuedTime),
+                    onProcessEvent: eventArgs =>
                     {
-                        var partitionId = kvp.Key;
-                        List<DateTimeOffset> partitionTimestamps = kvp.Value;
-
-                        Assert.That(partitionTimestamps.Count, Is.GreaterThan(1), $"{ partitionId }: more time stamp samples were expected.");
-
-                        for (int index = 1; index < partitionTimestamps.Count; index++)
+                        if (eventArgs.Data != null)
                         {
-                            var elapsedTime = partitionTimestamps[index].Subtract(partitionTimestamps[index - 1]).TotalSeconds;
-
-                            Assert.That(elapsedTime, Is.GreaterThan(maximumWaitTimeInSecs - 0.1), $"{ partitionId }: elapsed time between indexes { index - 1 } and { index } was too short.");
-                            Assert.That(elapsedTime, Is.LessThan(maximumWaitTimeInSecs + 5), $"{ partitionId }: elapsed time between indexes { index - 1 } and { index } was too long.");
-
-                            ++index;
+                            Interlocked.Increment(ref receivedEventsCount);
+                            if (receivedEventsCount == expectedEventsCount)
```

Comment:
Should we go with `>=` here just to be sure that we don't get hung if there's an extra increment?   _(the intent is to allow the test to move forward and fail in the final assertion rather than get caught up waiting for the cancellation timeout)_

Reply:
I almost went back and changed this earlier, but you convinced me :)